packages/base-controller/src/BaseControllerV2.ts

```diff
@@ -65,7 +67,13 @@ export interface StatePropertyMetadata<T extends Json> {
 export class BaseController<
   N extends string,
   S extends Record<string, Json>,
-  messenger extends RestrictedControllerMessenger<N, any, any, string, string>,
+  messenger extends RestrictedControllerMessenger<
```

Comment:
This is interesting. This would be a breaking change, and so I am wary of changing this. That said, I don't understand why we chose to use `any` here. I would be curious to hear if Mark can remember any context around this.

Reply:
`ActionConstraint` and `EventConstraint` in this PR are written to be the universal supertypes of their respective categories, so I don't think we'd be adding a meaningful constraint by moving away from `any`. Also no tests are being broken by this change, although I guess we'll have to check whether any of the other repos are using incompatible types.

Reply:
In general, using `any` in this context is not harmful in the same way that it is in other contexts. Types used in a generic constraints don't get applied to any specific variables, they're just constraints (i.e. by using `any` here, absolutely nothing is actually given the type `any`, even indirectly). For this reason I've sometimes put less effort into avoiding the use of `any` for generic constraints.

That said, more detailed constraints don't hurt either. And in this case, these constraints are already present in the `RestrictedControllerMessenger` type so this change seems like it would not affect anything functionally.

Reply:
Generic constraints are also helpful for intellisense hints and self-documenting code. If finding the right constraints isn't a significant blocker for actual feature development, I think it's worth the extra effort, even if it's not strictly necessary in terms of functionality.